custom/icds_reports/tasks.py

```diff
@@ -1697,19 +1697,14 @@ def reconcile_data_not_in_ucr(reconciliation_status_pk):
     status_record = UcrReconciliationStatus.objects.get(pk=reconciliation_status_pk)
     number_documents_missing = 0
 
-    data_not_in_ucr = get_data_not_in_ucr(status_record)
+    data_not_in_ucr = get_valid_data_not_in_ucr(status_record)
     doc_ids_not_in_ucr = {data[0] for data in data_not_in_ucr}
-    known_bad_doc_ids = set(
-        InvalidUCRData.objects.filter(doc_id__in=doc_ids_not_in_ucr).values_list('doc_id', flat=True))
 
     # republish_kafka_changes
     # running the data accessor again to avoid storing all doc ids in memory
     # since run time is relatively short and does not scale with number of errors
     # but the number of doc ids will increase with the number of errors
-    for doc_id, doc_subtype, sql_modified_on in get_data_not_in_ucr(status_record):
```

Comment:
Assuming it was an oversight to call this `get_data_not_in_ucr` twice, here as well as in line 1700

Reply:
`get_data_not_in_ucr` returns a generator, don't you need to convert it to a list so you can iteratee over it twice?